src/TestSuite/Fixture/SchemaLoader.php

```diff
@@ -29,77 +31,79 @@
  * support multiple database platforms. You should use migrations
  * for that instead.
  */
-class SchemaManager
+class SchemaLoader
 {
+    use InstanceConfigTrait;
+
     /**
      * @var \Cake\Console\ConsoleIo
      */
     protected $io;
 
+    /**
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'dropTables' => true,
+        'outputLevel' => ConsoleIo::NORMAL,
+    ];
+
     /**
      * SchemaManager constructor.
      *
-     * @param bool|null $verbose Output CLI messages.
+     * @param array<string, mixed> $config Config settings
      */
-    final public function __construct(?bool $verbose = false)
+    public function __construct(array $config = [])
     {
+        $this->setConfig($config);
+
         $this->io = new ConsoleIo();
-        $this->io->level($verbose ? ConsoleIo::NORMAL : ConsoleIo::QUIET);
+        $this->io->level($this->getConfig('outputLevel'));
     }
 
     /**
      * Import the schema from a file, or an array of files.
      *
-     * This function will drop all tables in the database and then
-     * load the provided schema file(s).
-     *
-     * @param string $connectionName Connection
-     * @param array<string>|string $file File to dump
-     * @param bool|null $verbose Set to true to display messages
-     * @param bool|null $enableDropping Will drop all tables prior to creating the schema (true by default)
+     * @param array<string>|string $files Schema files to load
+     * @param string $connectionName Connection name
+     * @param bool $dropTables Drop all tables prior to loading schema files
+     * @param bool $truncateTables Truncate all tables after loading schema files
      * @return void
-     * @throws \Exception if the truncation failed
-     * @throws \RuntimeException if the file could not be processed
      */
-    public static function create(
+    public function loadFiles(
+        $files,
         string $connectionName,
```

Comment:
Not sure if this should take a connection name. This and SchemaLoader is littered with calls to get the connection.

Reply:
Connection names is less work for the developer and all of this code is coupled to `Connection` already.